### PR TITLE
Fix a bug that Japanese files were not displayed correctly.

### LINKED ISSUE #9842: astro build can't handle images with Chinese paths
### Astro Info

```block
Astro                    v4.2.5
Node                     v20.11.0
System                   macOS (x64)
Package Manager          unknown
Output                   static
Adapter                  none
Integrations             @astrojs/starlight
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

Create a folder with a Chinese path under the 'assets' directory and place the image '测试.jpg' in it. The image is referenced by a relative path in 'content/docs'. It displays correctly in `astro dev`, but `astro build && astro preview` cannot locate this image.

`astro dev`
image

`astro build && astro preview`
image

### What's the expected result?

I hope that images in Chinese paths can be displayed normally after 'astro build', just like they do in 'astro dev'.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-gzbaza-bwvrxd?file=src%2Fcontent%2Fdocs%2F%E6%B5%8B%E8%AF%95%2F%E6%B5%8B%E8%AF%95.md

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by ktym4a**
fix #9842 

## Changes
Decode `node.properties.src` and to be the same as `file.data.imagePaths` value.

## Testing
I ran `pnpm --filter @example/blog run dev` and tested that the files were displayed correctly.

## Docs

I don't think an update is necessary.

### CODE REVIEW

File: .changeset/cool-colts-watch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@astrojs/markdown-remark": patch
+---
+
+Fixed a bug that Japanese file name is not displayed when using relative paths in markdowns.
```
Comment (ematipico) [inline .changeset/cool-colts-watch.md:5]:
```suggestion
Fixes a bug where non-UTF-8 file names are not displayed when using relative paths in markdowns.
```

**Review by ematipico — APPROVED** [review]
The fix seems sound to me

**Comment by ematipico** [comment]
Can you add a new test case, please? 

You want to add a new asset here: https://github.com/withastro/astro/tree/main/packages/astro/test/fixtures/astro-assets-prefix/ and update the `markdown.md` page. 

Add a new test case in this file: https://github.com/withastro/astro/blob/f27f790b922247c168eb167394cea8dd4c3e3390/packages/astro/test/astro-assets-prefix.test.js#L4

**Comment by ktym4a** [comment]
@ematipico
Thanks! I wanted to add a test case but didn't know where to put it.

and now Add test case.
I have also checked that the test fail before fix.

Is this test case clear?

**Comment by ematipico** [comment]
> @ematipico Thanks! I wanted to add a test case but didn't know where to put it.
> 
> and now Add test case. I have also checked that the test fail before fix.
> 
> Is this test case clear?

That's very clear, thank you!

**Pull request merged**